transport_nantes/mobilito/views.py

```diff
@@ -436,3 +443,41 @@ def get(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
         context = self.get_context_data(**kwargs)
         self.check_view_permission(context["mobilito_session"])
         return self.render_to_response(context)
+
+
+@csrf_protect
+def report_session(request: HttpRequest, **kwargs) -> HttpResponse:
+    """Report a session as inappropriate"""
+
+    if request.method == 'POST':
+        if not request.user.is_authenticated:
+            return HttpResponse(status=401)
+        session_sha1 = kwargs['session_sha1']
+        try:
+            session_object = get_object_or_404(Session, session_sha1=session_sha1)
+        except Http404:
+            logger.error(f"{request.user.email} tried to report a "
+                         "non-existing session")
+            raise Http404
+        if session_object:
+            _, created = SessionReport.objects.get_or_create(
+                session=session_object,
+                reporter_user=request.user,
+                reporter_session_id=request.session.session_key,
+                defaults={
+                    'session': session_object,
+                    'reporter_user': request.user,
+                    'reporter_session_id': request.session.session_key,
+                    'textual_report': request.POST.get('report-abuse-text')
+                })
+            if created:
+                logger.info(f"{request.user} reported session "
+                            f"{session_object.id}")
+            else:
+                logger.info(f"{request.user} tried to report session "
```

Comment:
This also deprives us of quite a lot of information.  A page that is reported multiple times is a stronger signal, not something about which we should fret.

Reply:
Multiple users can report the same session. 
A single user can only report once.

Reply:
Can you think of other sites that have actions you can/should only do once?  How do they communicate to you that you've done the action and can't do it again?

How do they know who you are and why do you think they make those decisions?